src/pipecat_flows/actions.py

```diff
@@ -68,6 +76,12 @@ def __init__(self, task: PipelineTask, flow_manager: "FlowManager", tts=None):
         # Register built-in actions
         self._register_action("tts_say", self._handle_tts_action)
         self._register_action("end_conversation", self._handle_end_action)
+        self._register_action("function", self._handle_function_action)
+
+        @self.task.event_handler("on_frame_reached_downstream")
```

Comment:
you can also refer to `@task.event_handler` directory since it's available here.